benches/benches/block_target_gas_set/contract.rs

```diff
@@ -1,20 +1,741 @@
+};
+use fuel_core_types::{
+    fuel_types::{
+        Address,
+        Word,
+    },
+    fuel_vm::consts::WORD_SIZE,
+};
+
+// BAL: Balance of contract ID
+// BHEI: Block height
+// BHSH: Block hash
+// BURN: Burn existing coins
+// CALL: Call contract
+// CB: Coinbase address
+// CCP: Code copy
+// CROO: Code Merkle root
+// CSIZ: Code size
+// LDC: Load code from an external contract
+// LOG: Log event
+// LOGD: Log data event
+// MINT: Mint new coins
+// RETD: Return from context with data
+// RVRT: Revert
+// SMO: Send message to output
+// SCWQ: State clear sequential 32 byte slots
+// SRW: State read word
+// SRWQ: State read sequential 32 byte slots
+// SWW: State write word
+// SWWQ: State write sequential 32 byte slots
+// TIME: Timestamp at height
+// TR: Transfer coins to contract
+// TRO: Transfer coins to output
+pub fn run_contract(group: &mut BenchmarkGroup<WallTime>) {
+    let contract_id = ContractId::zeroed();
+    let (mut service, rt) = service_with_contract_id(contract_id);
+    let mut rng = rand::rngs::StdRng::seed_from_u64(2322u64);
+    let asset_id = AssetId::zeroed();
+    let contract_id = ContractId::zeroed();
+    let script_data = script_data(&contract_id, &asset_id);
+
+    // bal contract
+    {
+        let contract_instructions =
+            vec![op::bal(0x13, 0x11, 0x10), op::jmpb(RegId::ZERO, 0)];
+
+        let mut instructions = setup_instructions();
+        instructions.extend(vec![op::call(0x10, RegId::ZERO, 0x11, 0x12)]);
+
+        replace_contract_in_service(&mut service, &contract_id, contract_instructions);
+
+        let id = "contract/bal contract";
+        run_with_service(
+            id,
+            group,
+            instructions,
+            script_data.clone(),
+            &service,
+            contract_id,
+            &rt,
+            &mut rng,
+        );
+    }
+    {
+        let mut instructions = setup_instructions();
+        instructions.extend(vec![op::bal(0x13, 0x11, 0x10), op::jmpb(RegId::ZERO, 0)]);
+
+        let id = "contract/bal script";
+        run_with_service(
+            id,
+            group,
+            instructions,
+            script_data.clone(),
+            &service,
+            contract_id,
+            &rt,
+            &mut rng,
+        );
+    }
+
+    // bhei
+    run(
+        "contract/bhei",
+        group,
+        vec![op::bhei(0x10), op::jmpb(RegId::ZERO, 0)],
+        vec![],
+    );
+
+    // bhsh
+    run(
+        "contract/bhsh",
+        group,
+        vec![
+            op::movi(0x10, Bytes32::LEN.try_into().unwrap()),
+            op::aloc(0x10),
+            op::move_(0x10, RegId::HP),
+            op::bhsh(0x10, RegId::ZERO),
+            op::jmpb(RegId::ZERO, 0),
+        ],
+        vec![],
+    );
+
+    // burn
+    {
+        let contract = vec![op::burn(RegId::ONE, RegId::HP), op::jmpb(RegId::ZERO, 0)];
+        let mut instructions = setup_instructions();
+        // TODO: I don't know why we need these extra ops
```

Comment:
We do it because we want to have a zeroed sub id for the asset. You allocated 32 zero bytes and after you started to use it as a sub id